src/lib/logical_query_plan/jit_aware_lqp_translator.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include "lqp_translator.hpp"
+#include "operators/jit_operator/operators/jit_compute.hpp"
+#include "operators/jit_operator_wrapper.hpp"
+
+namespace opossum {
+
+/* This class can be used as a drop-in replacement for the LQPTranslator.
+ * The JitAwareLQPTranslator will try to translate multiple AbstractLQPNodes into a single JitOperatorWrapper, whenever
+ * that is possible and seems beneficial. Otherwise it will fall back to the LQPTranslator.
+ *
+ * It works in two steps:
+ * 1) Determine, if we can/should add a JitOperatorWrapper node here and which nodes we can replace:
+ *    Starting from the current node, we perform a BFS through the query tree. For each node we will determine, whether
+ *    it is jitable (based on the node's type and parameters). We will follow each branch of the tree until we hit a
+ *    non-jitable node. Since StoredTableNodes are not jitable, this is guaranteed to happen for all branches.
```

Comment:
How do you deal with StoredTableNodes? Why are they not jittable?

Reply:
They are implicitly jitted, since they provide the data to the operator pipeline. But no dedicated JitOperator instance is created for `StoredTableNodes`.